src/App.jsx

```diff
@@ -14,45 +19,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function sortGoodsBy(goodList, sortBy, isNeedReverse) {
+  const goodListCopy = [...goodList];
+
+  goodListCopy.sort((good1, good2) => {
+    switch (sortBy) {
+      case ('alpabetic'):
+        return good1.localeCompare(good2);
+
+      case ('length'):
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isNeedReverse) {
+    goodListCopy.reverse();
+  }
+
+  return goodListCopy;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [isChanged, setIsChanged] = useState(false);
```

Comment:
actually you can avoid this additional state
```suggestion
const isShowResetButton = sortBy || isReversed;
```